src/coreclr/init-distro-rid.sh

```diff
@@ -154,7 +154,7 @@ initDistroRidGlobal()
         local distroRid=""
 
         # Check for musl-based distros (e.g Alpine Linux, Void Linux).
-        if (ldd --version 2>&1 || true) | grep -q musl; then
+        if "${rootfsDir}/usr/bin/ldd" --version 2>&1 | grep -q musl; then
```

Comment:
@am11 I think this is not going to work in cross compilation that we use for MUSL ARM64. The ldd in the rootfs is an ARM64 binary, so we cannot execute it. It seems we need to revert to the previous check of the /etc/os-release or parse the result of the `file` command ran on some known binary in the rootfs.

Reply:
@janvorli, it works because we are redirecting stderr to stdout, and stderr has musl symbol in it.

Reply:
For reference:

```
root@caf3ec0d0744:/# /crossrootfs/arm64/usr/bin/ldd --help
/crossrootfs/arm64/usr/bin/ldd: 2: exec: /lib/ld-musl-aarch64.so.1: not found
```